ssca-manager/service/src/main/java/io/harness/ssca/services/CdInstanceSummaryServiceImpl.java

```
@@ -177,21 +193,96 @@ public CdInstanceSummary getCdInstanceSummary(String accountId, String orgIdenti
+      return slsaVerificationSummaryBuilder.build();
+    }
+    try{
+      JsonNode executionNodeList = parseField(rootNode, "executionGraph", "nodeMap");
+      if(Objects.isNull(executionNodeList)){
+        return slsaVerificationSummaryBuilder.build();
+      }
+      for (JsonNode node : executionNodeList) {
+        if (node.get("stepType").asText().equals("SlsaVerification")) {
+          String identifier = getNodeValue(parseField(node, "identifier"));
+          JsonNode provenanceArtifactList = parseField(node, "outcomes", "artifact_container_" + identifier, "stepArtifacts", "provenanceArtifacts");
+          JsonNode provenanceArtifact = Objects.nonNull(provenanceArtifactList) ? provenanceArtifactList.get(0) : null;
+          slsaVerificationSummaryBuilder
+                  .slsaPolicyOutcomeStatus(getNodeValue(parseField(node, "outcomes", "policyOutput", "status")))
+                  .provenanceArtifact(getNodeValue(provenanceArtifact));
+          break;
+        }
+      }
+    }catch (Exception e){
+      log.error(String.format("Failed to extract SLSA Verification Data. Exception: %s",e));
+    }
+    return slsaVerificationSummaryBuilder.build();
+  }
+
+  private JsonNode getPmsExecutionSummary(Instance instance){
+    JsonNode rootNode = null;
+    try{
+      Object pmsExecutionSummary = NGRestUtils.getResponse(pipelineServiceClient.getExecutionDetailV2(
+              instance.getLastPipelineExecutionId(), instance.getAccountIdentifier(), instance.getOrgIdentifier(),
+              instance.getProjectIdentifier(), instance.getStageSetupId()));
+      rootNode = JsonUtils.asTree(pmsExecutionSummary);
+    } catch (Exception e){
+      log.error(String.format("PMS Request Failed. Exception: %s", e));
+    }
+    return rootNode;
+  }
+
+  private JsonNode parseField(JsonNode rootNode, String... path){
+      for (String field:path) {
+        try{
+          rootNode = rootNode.get(field);
+        }catch (Exception e){
```

Why are we doing it like this when we can have null checks here?

---

rootNode.get(field) will throw null pointer exception instead of returning null.